_posts/2021-02-01-aws-config-compliance-automation.md

```diff
@@ -0,0 +1,97 @@
+---
+layout:      posts
+background:  shorterBackground
+title:       "Cloud Compliance Automation"
+subtitle:    ""
+date:        2021-02-01 9:00
+author:      "Larry Anderson"
+headerImg:  "/assets/images/posts/order-chaos.jpg"
+description: ""
+---
+
+## Compliance Automation using AWS Config & ServiceNow
+
+In large companies, often there are entire departments of IT staff whose sole purpose is to assist with ensuring compliance with many policies and regulations. Oftentimes this proves very burdensome, with hundreds of disparate deployments all wishing to operate within corporate bounds but needing oversight to help create visibility of non-compliant items and steps towards remediation.
+
+When I started thinking about writing on this topic, the 1987 film "RoboCop" came to mind...
+
+<iframe style="height:360px;width:640px" src="https://www.youtube.com/embed/LJxvlpbc6M4" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+
+Much like the Enforcement Droid 209 (ED209), organizations often desire to automate their response to security posture changes which are not desirable in the light of compliance rules.
+
+## AWS Config
+
+If your chosen deployment environment is Amazon Web Services (AWS), one of the ways to assist with compliance is by utilizing AWS Config. AWS touts it as "a service that enables you to assess, audit, and evaluate the configurations of your AWS resources" which "enables you to simplify compliance auditing, security analysis, change management, and operational troubleshooting".
+
+AWS Config works by defining rules, which encapsulate requirements from agreed policy or regulation. AWS Config provides some <a href="https://docs.aws.amazon.com/config/latest/developerguide/managed-rules-by-aws-config.html">managed rules</a> directly, but also provides the ability to define custom rules as AWS Lambda Functions. These are run periodically against AWS resources to determine if configurations are satisfactorily compliant or not.
+
+While AWS Config does allow for remediation to be automated through a combination with AWS Systems Manager Automation documents, automated remediation is not always a desirable outcome. Often, automated remediation of compliance violations can cause application outages if not handled correctly, and the outage impact can be larger than the impact of allowing the non-compliance to extend for a time while appropriate manual remediation occurs.
```

Comment:
`While AWS Config does allow for remediation to be automated through a combination with AWS Systems Manager Automation documents`
Would "a combination *of* AWS Systems Manager Automation documents" make more sense here?

Reply:
sounds good

Reply:
might just change "through a combination with" to "via"